src/chainstate/burn/operations/mod.rs

```diff
@@ -415,6 +430,78 @@ impl BlockstackOperationType {
             BlockstackOperationType::DelegateStx(ref mut data) => data.burn_header_hash = hash,
         };
     }
+
+    pub fn pre_stx_to_json(op: &PreStxOp) -> serde_json::Value {
+        json!({
+            "PreStx": {
+                "block_height": op.block_height,
+                "burn_header_hash": &op.burn_header_hash.to_hex(),
+                "output": stacks_addr_serialize(&op.output),
+                "txid": op.txid,
+                "vtxindex": op.vtxindex,
+            }
+        })
+    }
+
+    pub fn stack_stx_to_json(op: &StackStxOp) -> serde_json::Value {
+        json!({
+            "StackStx": {
+                "block_height": op.block_height,
+                "burn_header_hash": &op.burn_header_hash.to_hex(),
+                "num_cycles": op.num_cycles,
+                "reward_addr": op.reward_addr.clone().to_b58(),
+                "sender": stacks_addr_serialize(&op.sender),
+                "stacked_ustx": op.stacked_ustx,
+                "txid": op.txid,
+                "vtxindex": op.vtxindex,
+            }
+        })
+    }
+
+    pub fn transfer_stx_to_json(op: &TransferStxOp) -> serde_json::Value {
+        json!({
+            "TransferStx": {
+                "block_height": op.block_height,
+                "burn_header_hash": &op.burn_header_hash.to_hex(),
+                "memo": memo_serialize(&op.memo),
+                "recipient": stacks_addr_serialize(&op.recipient),
+                "sender": stacks_addr_serialize(&op.sender),
+                "transfered_ustx": op.transfered_ustx,
+                "txid": op.txid,
+                "vtxindex": op.vtxindex,
+            }
+        })
+    }
+
+    pub fn delegate_stx_to_json(op: &DelegateStxOp) -> serde_json::Value {
+        json!({
+            "DelegateStx": {
+                "block_height": op.block_height,
+                "burn_header_hash": &op.burn_header_hash.to_hex(),
+                "delegate_to": stacks_addr_serialize(&op.delegate_to),
+                "delegated_ustx": op.delegated_ustx,
+                "sender": stacks_addr_serialize(&op.sender),
+                "reward_addr": &op.reward_addr.as_ref().map(|(index, addr)| (index, addr.clone().to_b58())),
+                "txid": op.txid,
+                "until_burn_height": op.until_burn_height,
+                "vtxindex": op.vtxindex,
+            }
+
+        })
+    }
+
```

Comment:
Can you add a docstring here explaining why an explicit function for JSON serialization is necessary?  Specifically, the JSON serialization is lossy, and less efficient than the serialization mode used in the DB.